### PR TITLE
Add eligibility API and CLI

### PR DESCRIPTION
**Pull request by dadgar**

### CODE REVIEW

File: api/nodes_test.go [hunk 1]
```diff
@@ -208,6 +208,72 @@ func TestNodes_ToggleDrain(t *testing.T) {
...
+	// Toggle on
+	wm, err = nodes.ToggleEligibility(nodeID, true, nil)
+	if err != nil {
+		t.Fatalf("err: %s", err)
+	}
+	assertWriteMeta(t, wm)
+
+	// Check again
+	out, _, err = nodes.Info(nodeID, nil)
+	if err != nil {
```
Comment (schmichael) [inline api/nodes_test.go]: require.Nil (and require.Equal for the field checks)

File: command/agent/node_endpoint_test.go [hunk 2]
```diff
@@ -302,6 +302,57 @@ func TestHTTP_NodeDrain(t *testing.T) {
 	})
 }
 
+func TestHTTP_NodeEligble(t *testing.T) {
```
Comment (schmichael) [inline command/agent/node_endpoint_test.go]: Eligible

File: command/node_eligibility.go [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+		}
+	}
+
+	// Check if node exists
+	if len(nodeID) == 1 {
+		c.Ui.Error(fmt.Sprintf("Identifier must contain at least two characters."))
+		return 1
+	}
+
+	nodeID = sanatizeUUIDPrefix(nodeID)
```
Comment (schmichael) [inline command/node_eligibility.go]: sanitize

File: nomad/node_endpoint_test.go [hunk 4]
```diff
@@ -901,6 +901,97 @@ func TestClientEndpoint_Drain_Down(t *testing.T) {
 	})
 }
 
+func TestClientEndpoint_UpdateEligibility(t *testing.T) {
```
Comment (schmichael) [inline nomad/node_endpoint_test.go]: Can we add a test for the guard against making a draining node eligible for scheduling? That's a pretty important case to test as if we accidently break it, drains are very very broken.

Comment (schmichael) [reply nomad/node_endpoint_test.go]: Never mind! I see the test below. Love the indepth validation around this critical invariant.

File: nomad/structs/structs.go [hunk 5]
```diff
@@ -302,6 +303,13 @@ type NodeUpdateDrainRequest struct {
 	WriteRequest
 }
 
+// NodeUpdateEligibilityRequest is used for updating the scheduling	eligibility
```
Comment (schmichael) [inline nomad/structs/structs.go]: Fell asleep on the spacebar for a second

Comment (dadgar) [reply nomad/structs/structs.go]: Hm must be a UI thing? In code it is one space

**Pull request merged**